pkgs/misc/vim-plugins/overrides.nix

```diff
@@ -147,6 +148,15 @@ self: super: {
    '';
   });
 
+  deoplete-tabnine = super.deoplete-tabnine.overrideAttrs(old: {
+    postPatch = ''
+      substituteInPlace rplugin/python3/deoplete/sources/tabnine.py \
+        --replace "path = get_tabnine_path(binary_dir)" "path = ${lib.getBin tabnine}/bin/TabNine"
+    '';
+    dependencies = with super; [ tabnine ];
```

Comment:
`dependencies` is for vim dependencies, no need to add tabnine there. The fact that you patched `tabnine.py` to refer to the tabnine package automatically makes it a dependency.